src/main/java/org/testng/internal/annotations/AnnotationHelper.java

```diff
@@ -345,4 +346,35 @@ private static String createMethodKey(Method m) {
 
     return result.toString();
   }
-}
+
+  static <A extends Annotation> A getAnnotationFromClass(Class<?> clazz, Class<A> a) {
+    try {
+      return clazz.getAnnotation(a);
+    } catch (Throwable t) {
+      String msg = "Encountered problems when parsing the annotation on class " + clazz.getCanonicalName() + ". ";
```

Comment:
Just check if `a` is not null because getAnnotation will throw an NPE which is not related to types.

Reply:
Fixed. Added null checks for both parameters of all the 3 new methods that were added.